### PR TITLE
Update Jenkins plugins

### PR DESCRIPTION
**Pull request by nuclearsandwich**
I reviewed every plugin installed for updates to their changelogs and noted possible breaking changes where I could see them.
I also added a number of TODO comments over plugins which I don't think we're directly using as well as plugins that are still listed only as possible implicit dependencies due to plugins that express dependencies on older versions of jenkins core where these plugins were part of the core.

* PrioritySorter 4.0.0 -> 4.1.0
  * bug fixes only
* analsysis-model-api 10.0.0 -> 10.9.1
  * New features, bug fixes, and internal changes
* ant 1.9 -> 1.13
  * dependency updates and bug fixes
* antisamy-markup-formatter 2.1 -> 2.7
  * internal dependency changes and bug fixes
* audit-trail 3.7 -> 3.10
  * bug fixes and new features
* badge 1.6 -> 1.9.1
  * changelog incomplete
  * addresses security fixes according to plugin update center
* bootstrap5-api ADDED -> 5.1.3-4
  * dependency of forensics, data-tables-api, warnings-ng
* bouncycastle-api 2.17 -> 2.17
  * bug fixes and dependency updates
* branch-api 2.6.3 -> 2.7.0
  * Refactor and changes of Organization features not used by ros_buildfarm
  * Removed CustomOrganizationFolderDescriptor (unused by ros_buildfarm)
* caffeine-api 2.9.1-23.v51c4e2c879c8 -> 2.9.2-29.v717aac953ff3
  * bug fixes and dependency updates
* checks-api 1.7.0 -> 1.7.2
  * bug fixes and dependency updates
* cloudbees-folder 6.15 -> 6.17
  * new features and dependency updates
* command-launcher 1.2 -> 1.6
  * dependency updates
* conditional-buildstep 1.3.6 -> 1.4.1
  * bug fix (table-to-div migration)
* configuration-as-code 1.51 -> 1346.ve8cfa_3473c94
  * New features like reloading and merge strategies
  * security updates
* copyartifact 1.45.2 -> 1.46.2
  * new features and bug fixes
* credentials 2.4.1 -> 1074.v60e6c29b_b_44b_
  * new features and bug fixes
  * a regression was introduced in 2.6.x and fixed in 1055.v1346ba467ba1
* credentials-binding 1.24 -> 1.27.1
  * new features and bug fixes
* dashboard-view 2.16 -> 2.18
  * new features and dependency updates
* data-tables-api 1.10.23-3 -> 1.11.3-6
  * breaking change! migration to bootstrap 5.
    need to examine which plugins depend on this and their requirements.
* display-url-api 2.3.4 -> 2.3.5
  * bug fixes
* durable-task 1.27 -> 493.v195aefbb0ff2
  * breaking change! [x] requires Jenkins >= 2.277.1
* echarts-api 5.1.0-2 -> 5.2.2-2
  * bug fixes and dependency updates
* email-ext 2.78 -> 2.87
  * removed support for analysis-core-plugin (no longer used by ros_buildfarm)
* embeddable-build-status 2.0.2 -> 2.0.3
  * no changelong entry
* external-monitor-job 1.7 -> 191.v363d0d1efdf8
  * unused by ros_buildfarm, an implied/assumed dependency of some older plugins
* extra-columns 1.23 -> 1.25
  * bug fixes
  * updated for icon removal in core
* font-awesome-api 5.15.2-2 -> 5.15.4-5
  * bug fixes and dependency updates
* forensics-api 1.0.0 -> 1.7.0
  * bug fixes, improvements, and dependency updates
* ghprb 1.42.1 -> 1.42.2
  * adds configuration as code support
  * dependency updates
* git 4.7.1 -> 4.10.3
  * security fixes
  * bug fixes, improvements, dependency updates
* git-client 3.7.1 -> 3.11.0
  * bug fixes, improvements, dependency updates
* git-forensics 1.0.0 -> 1.3.0
  * bug fixes, improvements, dependency updates
* git-server 1.9 -> 1.10
  * improvements
* github 1.29.5 -> 1.34.1
  * improvements, bug fixes, dependency updates
* github-api 1.95 -> 1.301-378.v9807bd746da5
  * improvements, bug fixes, dependency updates
  * unbundles okhttp3 probably needs a new plugin for it
* github-branch-source 2.5.8 -> 2.11.4
  * new features, bug fixes, and dependency updates
* github-oauth 0.33 -> 0.37
  * new features, bug fixes, and dependency updates
  * breaking change! Team slugs used for authentication instead of team names. We have always used slug-compatible names so should be unaffected.
  * Propagate to release notes of chef cookbook version.
* greenballs 1.15 -> REMOVED
  * made effectively obsolete by jenkins UI updates
* groovy 2.2 -> 2.4
  * minor fixes and test coverage improvements
* groovy-postbuild 2.4.3 -> 2.5
  * bug fixes, improvements, and dependency updates
* htmlpublisher 1.21 -> 1.29
  * updates minimum jenkins version (to 2.250.3)
  * bug fixes
* jackson2-api 2.12.3 -> 2.13.1-246.va8a9f3eaf46a
  * bug fixes, new features, and dependency updates
* javax-activation-api ADDED -> 1.6.2-5
  * dependency of javax-mail-api
* javax-mail-api ADDED -> 1.6.2-5
  * dependency of mailer
* javadoc 1.4 -> 1.6
  * new features
* jaxb ADDED -> 2.3.0.1
  * dependency of xunit 3.0.5
* jdk-tool 1.1 -> 1.5
  * unused by ros_buildfarm, an implied/assumed dependency of some older plugins
* jjwt-api ADDED -> 0.11.2-9.c8b45b8bb173
  * dependency of github-branch-source-plugin
* jobConfigHistory 2.18.3 -> 2.31-rc1107.2354f08725a_8
  * new features and dependency updates
* jnr-posix-api ADDED -> 3.1.7-1
  * dependency of pam plugin
* jquery 1.12.4-0 -> 1.12.4-1
  * bug fixes, new features, and dependency updates
* jquery3-api 3.6.0-1 -> 3.6.0-2
  * dependency updates
* junit 2.49 -> 1.54
  * bug fixes, new features, and dependency updates
  * 1.55 requires newer than LTS Jenkins.
* ldap 1.26 -> 2.7
  * unused plugin
* log-parser 2.1 -> 2.2
  * bug fixes
* mailer 1.32.1 -> 408.vd726a_1130320
  * dependency updates
  * security fixes
* matrix-auth 2.6.6 -> 3.0.1
  * Migration of permissions to new format
* matrix-project 1.18 -> 1.20
  * updates permissions format to disambiguate users and groups.
    ambiguous format can be read but downgrading not possible after migrating to unambiguous format
  * adds configuration-as-code support
  * security fixes
* maven-plugin 3.4 -> 3.16
  * unused plugin
* mercurial 2.14 -> 2.16
  * bug fixes and dependency updates
  * updates for svg icons
* metrics 3.1.2.10 -> 4.0.2.8.1
  * update minimum jenkins version to 2.60 (lol)
  * improvements and fixes
  * security fixes
* modernstatus 1.2 -> REMOVED
  * replaced by jenkins UI updates
* monitoring 1.80.0 -> 1.90.0
  * security fixes in optional components
  * fixes and improvements
* okhttp-api ADDED -> 4.9.3-105.vb96869f8ac3a
  * New dependency of github-api
* pam-auth 1.5.1 -> 1.7
  * unused plugin
* parameterized-trigger 2.40 -> 2.43
  * new features and fixes
  * icon fixes
* pipeline-utility-steps 2.5.0 -> 2.12.0
  * Seems to be unused
  * bug fixes, improvements, and dependency updates
* plain-credentials 1.7 -> 1.8
  * maintenance release for JDK11 compat
* plot 2.1.6 -> 2.1.10
  * new features, bug fixes, dependency updates
  * icon fixes
* plugin-util-api 2.1.0 -> 2.13.0
  * improvements, bug fixes, dependency updates
* prism-api ADDED -> 1.25.0-2
  * dependency of warnings-ng
* popper2-api ADDED -> 2.11.2-1
  * New dependency of github-api
* rebuild 1.31 -> 1.33
  * bug fixes and dependency updates
* run-condition 1.2 -> 1.5
  * new features and dependency updates
  * tables-to-divs fixes
* scm-api 2.6.4 -> 595.vd5a_df5eb_0e39
  * bug fixes and dependency updates
* script-security 1.76 -> 1131.v8b_b_5eda_c328e
  * new features and fixes
* snakeyaml-api 1.27.0 -> 1.29.1
  * bump snakeyaml version
* ssh-agent 1.17 -> 1.24.1
  * bug fixes and dependency updates
  * breaking change! deleting non-exec based agent factories
    This needs to be checked https://github.com/jenkinsci/ssh-agent-plugin/pull/48
* ssh-credentials 1.18.1 -> 1.19
  * new features and dependency updates
* ssh-slaves 1.28.1 -> 1.33.0
  * possibly not used any longer
  * bug fixes and dependency updates
  * Bump jenkins core requirement to 2.190
  * breaking change! removed sftp client
* sshd ADDED -> 3.1.0
* structs 1.24 -> 308.v852b473a2b8c
  * dependency updates
* subversion 2.14.0 -> 2.15.2
  * bug fixes and dependency updates
  * security fixes
* swarm NOT UPDATED
  * TODO update swarm plugin separately in tandem with agent refreshes.
* timestamper 1.11.3 -> 1.11.3
  * configuration-as-code support
  * design fixes for sidebar
  * bug fixes and dependency updates
* token-macro 2.15 -> 267.vcdaea6462991
  * dependency updates
  * jdk11 readiness updates
* variant ADDED -> 1.4
  * dependency of metrics plugin
* warnings-ng 9.0.1 -> 9.11.0
  * bug fixes, improvements, and dependency updates
* windows-slaves 1.3.1 -> 1.8
  * possibly not used any longer
  * bug fixes, improvements, and dependency updates
* workflow-api 2.42 -> 1136.v7f5f1759dc16
  * bug fixes and dependency updates
* workflow-cps 2.90 -> 2648.va9433432b33c
  * improvements, bug fixes and dependency updates
* workflow-cps-global-lib 2.15 -> 552.vd9cc05b8a2e1
  * improvements, bug fixes and dependency updates
* workflow-job 2.40 -> 2.42
  * improvements, bug fixes and dependency updates
* workflow-multibranch 2.23 -> 706.vd43c65dec013
  * improvements, bug fixes and dependency updates
  * svg icons
* workflow-scm-step 2.12 -> 2.13
  * no documented changes
* workflow-step-api 2.23 -> 622.vb_8e7c15b_c95a_
  * improvements, bug fixes and dependency updates
* workflow-support 3.8 -> 813.vb_d7c3d2984a_0
  * improvements, bug fixes and dependency updates
* xunit 3.0.2 -> 3.0.5
  * no documented changes

### CODE REVIEW

**Comment by gavanderhoorn** [comment]
Did you run through the changelogs of all these plugins?

That's got to have taken some time :tired_face:

**Comment by nuclearsandwich** [comment]
> Did you run through the changelogs of all these plugins?

Yep. Reviewed each plugin's changelog as I was going through. I missed a few plugin dependencies so I've force-pushed this commit a few times over. I was working on a tool to make compiling a PR from standalone commits easier but it's harder to manage when there are inter-dependencies between plugins that require multiple updates for an atomic commit.

As a result it's best to just update the entire system in tandem to avoid intermediate states that render Jenkins unbootable and thus can't be recovered from. Things should get a lot better with the use of the [plugin-manager](https://github.com/jenkinsci/plugin-installation-manager-tool) but I want to implement that directly in the [jenkins](https://github.com/sous-chefs/jenkins) cookbook of which I'm a maintainer-errant.

> That's got to have taken some time :tired_face:

basically that was my day yesterday.

**Comment by nuclearsandwich** [comment]
There were a couple of plugins where I put their changes in the changelog but missed bumping the version. That's fixed in the latest push.

The `junit` and `workflow-job` plugins can't be upgraded to their latest version due to our Jenkins version as they don't currently target the LTS release and I am reserving the swarm plugin update for a dedicated PR as I will also need to test that with the agents.
